libraries/AP_HAL_SITL/SITL_cmdline.cpp

```diff
@@ -232,6 +232,9 @@ void SITL_State::_parse_command_line(int argc, char * const argv[])
             break;
         case 's':
             speedup = strtof(gopt.optarg, nullptr);
+            char speedup_string[18];
+            snprintf(speedup_string, sizeof(speedup_string), "SIM_SPEEDUP=%s", gopt.optarg);
```

Comment:
Use `speedup` instead of gopt.optarg?

Reply:
it is simplier this way, otherwise with %f and speedup we got "warning: ‘%f’ directive output truncated writing between 8 and 317 bytes into a region of size 6 [-Wformat-truncation=]"
a more c++ way would be to use std::string and to_string